llvm/lib/Transforms/InstCombine/InstCombineCalls.cpp

```diff
@@ -1796,6 +1796,19 @@ Instruction *InstCombinerImpl::visitCallInst(CallInst &CI) {
     if (Instruction *NewMinMax = factorizeMinMaxTree(II))
        return NewMinMax;
 
+    // Try to fold minmax based on range information
+    ICmpInst::Predicate Pred =
+        ICmpInst::getNonStrictPredicate(MinMaxIntrinsic::getPredicate(IID));
+    bool IsSigned = MinMaxIntrinsic::isSigned(IID);
+    const auto LHS_CR = llvm::computeConstantRangeIncludingKnownBits(
+        I0, IsSigned, SQ.getWithInstruction(II));
```

Comment:
It looks like this still has some compile-time overhead. Would an early exit for full range help, to avoid computing the second one?

Reply:
Done.
Before: http://llvm-compile-time-tracker.com/compare.php?from=a863b121c7b28f2c30439ccbd70927ed481835cb&to=5297a541aa26eab6ac5e12e839e4931f08744671&stat=instructions:u
After: http://llvm-compile-time-tracker.com/compare.php?from=a863b121c7b28f2c30439ccbd70927ed481835cb&to=94c65a141a052df42508d40b0d6813721dba77fc&stat=instructions:u

I think it is enough to handle minmax with constant rhs. Then we can avoid computing `RHS_CR`.
See also https://github.com/dtcxzyw/llvm-opt-benchmark/pull/26/files.

Reply:
Yeah, just handling constants make sense.

Reply:
Done.